### PR TITLE
[tune] Cache MNIST and restore MNIST tests

### PR DESCRIPTION
**Pull request by richardliaw**
## Why are these changes needed?
MNIST tests are flaky and download takes forever. This caches the MNIST download across all builds with a public anyscale bucket.

cc @amogkam @krfricke

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by amogkam** [comment]
@richardliaw could we take in the data path as a command line arg for all these examples that defaults to "~/data"? That way the user can specify their own data paths without having to modify the code.

**Comment by richardliaw** [comment]
@amogkam can we do that in a later PR?

**Review by amogkam — APPROVED** [review]
OK that sounds good

**Review by clarkzinzow — APPROVED** [review]
+1 to @amogkam's suggestion, but agreed that it can wait for a future PR.

File: ci/travis/install-dependencies.sh [hunk 1]
```diff
@@ -257,6 +257,12 @@ install_toolchains() {
   fi
 }
 
+download_mnist() {
+  mkdir -p "${HOME}/data"
+  curl -o "${HOME}/data/mnist.zip" https://ray-ci-mnist.s3-us-west-2.amazonaws.com/mnist.zip
+  unzip "${HOME}/data/mnist.zip" -d "${HOME}/data"
```
Comment (richardliaw) [inline ci/travis/install-dependencies.sh:263]: does this look OK? we're hosting this ourselves and the bucket has public read access

**Review by simon-mo — APPROVED** [review]

File: ci/travis/install-dependencies.sh (hunk 1 above)
Comment (simon-mo) [reply ci/travis/install-dependencies.sh:263]: yeah this is totally fine

**Pull request merged**